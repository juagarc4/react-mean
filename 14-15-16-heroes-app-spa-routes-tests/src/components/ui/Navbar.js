import React, { useContext } from 'react'
import { NavLink, useHistory } from 'react-router-dom'
import { AuthContext } from 'auth/AuthContext'
import { types } from 'types/types'

export const Navbar = () => {
  const {
    user: { name },
    dispatch,
  } = useContext(AuthContext)
  const history = useHistory()
  const handleLogout = () => {
    dispatch({
      type: types.logout,
    })
    history.replace('/login')
  }
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <div className="container-fluid">
        <NavLink className="navbar-brand" to="/">
          HeroesApp
        </NavLink>
        <button
          className="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbar"
          aria-controls="navbar"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbar">
          <div className="navbar-nav me-auto mb-2 mb-lg-0">
            <NavLink
              activeClassName="active"
              className="nav-item nav-link"
              exact
              to="/marvel"
            >
              Marvel
            </NavLink>

            <NavLink
              activeClassName="active"
              className="nav-item nav-link"
              exact
              to="/dc"
            >
              DC
            </NavLink>

            <NavLink
              activeClassName="active"
              className="nav-item nav-link"
              exact
              to="/search"
            >
              Search
            </NavLink>
            <hr />
            <span className="nav-item nav-link text-info">{name}</span>
            <button className="nav-item nav-link btn" onClick={handleLogout}>
              Logout
            </button>
          </div>
        </div>
      </div>
    </nav>
  )
}
