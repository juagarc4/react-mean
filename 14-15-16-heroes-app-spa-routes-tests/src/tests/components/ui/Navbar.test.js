import { AuthContext } from 'auth/AuthContext'
import { mount } from 'enzyme'
import { Navbar } from 'components/ui/Navbar'
import { MemoryRouter, Router } from 'react-router-dom'
import { types } from 'types/types'

describe('Tests in <Navbar/>', () => {
  const historyMock = {
    push: jest.fn(),
    replace: jest.fn(),
    location: {},
    listen: jest.fn(),
    createHref: jest.fn(),
  }
  const contextValue = {
    dispatch: jest.fn(),
    user: { name: 'Raul', logged: true },
  }

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter>
        <Router history={historyMock}>
          <Navbar />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  )

  afterEach(() => jest.clearAllMocks())

  test('Should render correctly ', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.text-info').text().trim()).toBe('Raul')
  })
  test('Should call logout and useHistory', () => {
    wrapper.find('.btn-login').simulate('click')
    expect(contextValue.dispatch).toHaveBeenCalledWith({ type: types.logout })
    expect(historyMock.replace).toHaveBeenCalledWith('/login')
  })
})
