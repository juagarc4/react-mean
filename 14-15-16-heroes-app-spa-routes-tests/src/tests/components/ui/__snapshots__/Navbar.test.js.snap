// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Navbar/> Should render correctly  1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <a
      aria-current={null}
      className="navbar-brand"
      onClick={[Function]}
    >
      HeroesApp
    </a>
    <button
      aria-controls="navbar"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbar"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbar"
    >
      <div
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          Search
        </a>
        <hr />
        <span
          className="nav-item nav-link text-info"
        >
          Raul
        </span>
        <button
          className="nav-item nav-link btn btn-login"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</nav>
`;
